LMS/Billings/Repositories/Providers/ProviderRepository.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace LMS\Billings\Repositories\Providers;
+
+use LMS\Billings\Models\Provider;
+
+class ProviderRepository
+{
+    private Provider $model;
+
+    public function __construct()
+    {
+        $this->model = new Provider();
+    }
+
+    public function activateProvider(Provider $provider): bool
+    {
+        foreach ($this->model->all() as $providerItem) {
+            $providerItem->turnOff();
+        }
```

Comment:
Isole este foreach em outra função.

Reply:
Beleza, Arthur? a titulo de dúvida, porque seria melhor outra função?

Reply:
criar uma função com o nome mais claro do que está acontecendo. por exemplo: $this->turnOffAllProviders();